builder/Dockerfile

```diff
@@ -14,7 +14,7 @@ RUN dnf -y update \
         binutils-devel \
         bison \
         ca-certificates \
-        clang \
+        clang-16.0.1 \
```

Comment:
Can we have a comment somewhere appropriate that indicates why we need this specific version (possibly with the bugzilla links as well)

Reply:
I'll add it after this CI run, so we can confirm the fix works properly (just 2 more hours to go! 🤦🏻‍♂️)